event.c

```diff
@@ -676,17 +676,17 @@ static void event_image_preview(struct context *cnt,
 
         if ((cnt->ffmpeg_output || (cnt->conf.movie_extpipe_use && cnt->extpipe)) && !use_imagepath) {
             if (cnt->conf.movie_extpipe_use && cnt->extpipe) {
-                basename_len = strlen(cnt->extpipefilename) + 1;
-                strncpy(previewname, cnt->extpipefilename, basename_len);
+                basename_len = MIN(strlen(cnt->extpipefilename) + 1, sizeof(previewname) - 1);
+                strncpy(previewname, cnt->extpipefilename, sizeof(previewname));
                 previewname[basename_len - 1] = '.';
```

Comment:
Sorry I didn't notice this in my first review. This is better implemented using `snprintf` to take care of the file extension string:

    snprintf(previewname, sizeof(previewname), "%s.%s", cnt->extpipefilename, imageext(cnt));

And thus, the `basename_len` line before and `previewname` line after can both be removed as unnecessary.